components/rating/stories/template.js

```diff
@@ -12,30 +12,89 @@ export const Template = ({
 	max = 5,
 	value = 0,
 	isReadOnly = false,
-	isFocused = false,
+	isKeyboardFocused = false,
 	isDisabled = true,
 	isEmphasized = false,
 	customClasses = [],
 	id = getRandomId("rating"),
 } = {}, context = {}) => {
 	const { updateArgs } = context;
+	document.addEventListener("DOMContentLoaded", function() {
+		const rating = document.getElementById(id);
+		if (!rating) return;
+		if (rating.classList.contains("is-disabled") || rating.classList.contains("is-readOnly")) return;
+		const icons = Array.from(rating.getElementsByClassName("spectrum-Rating-icon"));
+		let hoverIndex = -1;
+		let selectedIndex = -1;
+
+		const updateHoverState = () => {
+			icons.forEach((icon, index) => {
+				const activeStar = icon.querySelector(".spectrum-Rating-starActive");
+				const inactiveStar = icon.querySelector(".spectrum-Rating-starInactive");
+
+				if (index <= hoverIndex ||
+					(index <= selectedIndex && hoverIndex === -1)
+				) {
+					icon.classList.add("is-hoverSelection");
+					activeStar.style.display = "block";
+					inactiveStar.style.display = "none";
+				}
+				else {
+					icon.classList.remove("is-hoverSelection");
+					activeStar.style.display = "none";
+					inactiveStar.style.display = "block";
+				}
+			});
+		};
+
+		rating.addEventListener("mouseleave", function() {
+			icons.forEach(icon => {
+				icon.classList.remove("is-hoverSelection");
+				icon.querySelector(".spectrum-Rating-starActive").style.display = "";
+				icon.querySelector(".spectrum-Rating-starInactive").style.display = "";
+			});
+		});
+
+		icons.forEach((icon, index) => {
+			if (icon.classList.contains("is-selected")) selectedIndex = index;
+
+			icon.addEventListener("mouseover", function() {
+				hoverIndex = index;
+				updateHoverState();
+			});
+
+			icon.addEventListener("mouseleave", function(event) {
+				if (!rating.contains(event.relatedTarget)) {
```

Comment:
I'm not sure I've seen `relatedTarget`. Could you explain what this `if` is checking for? (I'll have to look it up!)